### PR TITLE
DBZ-1141 Correct default falback values for connect date types

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1141

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@jpechane, there seems to be some overlap with https://github.com/debezium/debezium/pull/787; probably it makes sense to rebase this one here onto #787, it seems you also added the integration test I had asked about in the other one?

**Comment by jpechane** [comment]
@gunnarmorling I've added `money` and `bit` datatype from the refrenced PR.

File: debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java [hunk 1]
```diff
@@ -474,7 +483,7 @@ protected Object convertTimestampToEpochNanos(Column column, Field fieldDefn, Ob
      */
     protected Object convertTimestampToEpochMillisAsDate(Column column, Field fieldDefn, Object data) {
         // epoch is the fallback value
-        return convertValue(column, fieldDefn, data, 0L, (r) -> {
+        return convertValue(column, fieldDefn, data, connectDateTypesUsed() ? new java.util.Date(0) : 0L, (r) -> {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java]: I don't think this check is needed here, it has been done by the caller before. This method always produces `Date`, also see its name.

File: debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java [hunk 2]
```diff
@@ -574,7 +583,7 @@ protected Object convertTimeToNanosPastMidnight(Column column, Field fieldDefn,
      */
     protected Object convertTimeToMillisPastMidnightAsDate(Column column, Field fieldDefn, Object data) {
         // epoch is the fallback value
-        return convertValue(column, fieldDefn, data, 0L, (r) -> {
+        return convertValue(column, fieldDefn, data, connectDateTypesUsed() ? new java.util.Date(0) : 0L, (r) -> {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/jdbc/JdbcValueConverters.java]: Same here.

**Pull request closed without merging**